google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/BucketIamSnippets.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * EDITING INSTRUCTIONS
+ * This file is referenced in Storage's javadoc. Any change to this file should be reflected in
+ * Storage's javadoc.
+ */
+
+package com.google.cloud.examples.storage.snippets;
+
+import com.google.cloud.Identity;
+import com.google.cloud.Policy;
+import com.google.cloud.Role;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.storage.StorageRoles;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class contains Bucket-level IAM snippets for the {@link Storage} interface.
+ */
+public class BucketIamSnippets {
+
+  private final Storage storage;
+
+  public BucketIamSnippets(Storage storage) {
+    this.storage = storage;
+  }
+
+  /**
+   * Example of listing the Bucket-Level IAM Roles and Members
+   */
+
+  public Policy listBucketIamMembers(String bucketName) {
+    // [START view_bucket_iam_members]
+    // Initialize a Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // Get IAM Policy for a bucket
+    Policy policy = storage.getIamPolicy(bucketName);
+
+    // Print Roles and its identities
+    Map<Role, Set<Identity>> policyBindings = policy.getBindings();
+    for(Map.Entry<Role, Set<Identity>> entry : policyBindings.entrySet()) {
+        System.out.printf("Role: %s", entry.getKey());
+        System.out.printf(" Identities: %s\n", entry.getValue());
+    }
+    // [END view_bucket_iam_members]
+    return policy;
+  }
+
+  /**
+   * Example of adding a member to the Bucket-level IAM
+   */
+
+  public void addBucketIamMember(String bucketName, Role role, Identity identity) {
+    // [START add_bucket_iam_member]
+    // Initialize a Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // Get IAM Policy for a bucket
+    Policy policy = storage.getIamPolicy(bucketName);
+
+    // Add identity to Bucket-level IAM role
+    Policy updatedPolicy = policy.toBuilder().addIdentity(role, identity).build();
+
+    // Update the bucket IAM Policy
+    storage.setIamPolicy(bucketName, updatedPolicy);
+
+    System.out.printf("Added %s with role %s to %s\n", identity, role, bucketName);
```

Comment:
In this way the user would not know if the updated policy actually took effect.
It would be better to print the policy _returned from Storage_:
```java
...
Policy policy = storage.setIamPolicy(bucketName, updatedPolicy);
System.out.printf("Added %s with role %s to %s\n", policy.getBindings().get(role),...);
```

(Same comment for remove)